Tasks/SshV0/ssh.ts

```diff
@@ -66,8 +66,14 @@ async function run() {
                 tl.debug('No script header detected.  Adding: ' + bashHeader);
                 inlineScript = bashHeader + os.EOL + inlineScript;
             }
-            scriptFile = path.join(os.tmpdir(), 'sshscript_' + new Date().getTime()); // default name
+            const scriptFileDir = os.tmpdir();
+            scriptFile = path.join(scriptFileDir, 'sshscript_' + new Date().getTime()); // default name
             try {
+                // Make sure the directory exists or else we will get ENOENT
+                if (!fs.existsSync(scriptFileDir))
+                {
+                    tl.mkdirP(scriptFileDir);
+                }
                 fs.writeFileSync(scriptFile, inlineScript);
```

Comment:
I think this is fine for the short term, but how often are we going to be writing this code?  (where else is this code already?  Or, where else _should_ this code be?) Do we need to make a `tl.writeTempFile`?

Reply:
I think that's fair.  We're also on a really old version of task lib (0.8).  But presumably anywhere that's writing a temp file would have this bug.

Reply:
Actually, I wonder if we should just be using the agent's _temp directory.  I'll look into it.

Reply:
I think we should be doing both.  We should prefer `Agent.TempDirectory` if it is available (agent version 2.115).   There are also quite a few tasks that write temp files like this, so we should have a common function.

For this PR I'll switch to using `Agent.TempDirectory`.  I'll look into adding a common implementation later.